mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java

```diff
@@ -79,33 +84,57 @@ class ResourceClusterActor extends AbstractActorWithTimers {
 
     private final Duration heartbeatTimeout;
     private final Duration assignmentTimeout;
+    private final Duration disabledTaskExecutorsCheckInterval;
 
     private final Map<TaskExecutorID, TaskExecutorState> taskExecutorStateMap;
     private final Clock clock;
     private final Set<TaskExecutorID> taskExecutorsReadyToPerformWork;
     private final RpcService rpcService;
     private final ClusterID clusterID;
     private final MantisJobStore mantisJobStore;
+    private final Set<DisableTaskExecutorsRequest> activeDisableTaskExecutorsRequests;
+    private final JobMessageRouter jobMessageRouter;
 
-    static Props props(final ClusterID clusterID, final Duration heartbeatTimeout, Duration assignmentTimeout, Clock clock, RpcService rpcService, MantisJobStore mantisJobStore) {
-        return Props.create(ResourceClusterActor.class, clusterID, heartbeatTimeout, assignmentTimeout, clock, rpcService, mantisJobStore);
+    static Props props(final ClusterID clusterID, final Duration heartbeatTimeout, Duration assignmentTimeout, Duration disabledTaskExecutorsCheckInterval, Clock clock, RpcService rpcService, MantisJobStore mantisJobStore, JobMessageRouter jobMessageRouter) {
+        return Props.create(ResourceClusterActor.class, clusterID, heartbeatTimeout, assignmentTimeout, disabledTaskExecutorsCheckInterval, clock, rpcService, mantisJobStore, jobMessageRouter);
     }
 
     ResourceClusterActor(
         ClusterID clusterID,
         Duration heartbeatTimeout,
         Duration assignmentTimeout,
+        Duration disabledTaskExecutorsCheckInterval,
         Clock clock,
         RpcService rpcService,
-        MantisJobStore mantisJobStore) {
+        MantisJobStore mantisJobStore,
+        JobMessageRouter jobMessageRouter) {
         this.clusterID = clusterID;
         this.heartbeatTimeout = heartbeatTimeout;
         this.assignmentTimeout = assignmentTimeout;
+        this.disabledTaskExecutorsCheckInterval = disabledTaskExecutorsCheckInterval;
+
         this.clock = clock;
         this.rpcService = rpcService;
+        this.jobMessageRouter = jobMessageRouter;
         this.taskExecutorStateMap = new HashMap<>();
         this.taskExecutorsReadyToPerformWork = new HashSet<>();
         this.mantisJobStore = mantisJobStore;
+        this.activeDisableTaskExecutorsRequests = new HashSet<>();
+    }
+
+    @Override
+    public void preStart() throws Exception {
+        super.preStart();
+        List<DisableTaskExecutorsRequest> activeRequests =
+            mantisJobStore.loadAllDisableTaskExecutorsRequests(clusterID);
+        for (DisableTaskExecutorsRequest request : activeRequests) {
+            onNewDisableTaskExecutorsRequest(request);
+        }
+
+        timers().startPeriodicTimer(
+            "periodic-disabled-task-executors-test",
```

Comment:
nit: periodic-disabled-task-executors-test to periodic-disabled-task-executors-{clusterID}

Reply:
nice find!